.github/workflows/build-and-test-on-pr-events.yml

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License
+
+name: Pull Request Build and Test Workflow
+
+# Run this workflow every time a new pull request is created in the repository
+on:
+  push:
+    branches:
```

Comment:
So this workflow will also be triggered when pushing to these branches. Is this the intended behavior? Or you want it to be triggered only on pull request?

Reply:
Yes, I do intend to trigger this workflow whenever those branches are updated (a PR is merged to it or someone force push to it). 

If we do not include the `push` event here, will the workflow be trigger when we merge a PR to any of those branches?

Reply:
Yeah, then you may want to change the name / description here:

"name: Pull Request Build and Test Workflow

# Run this workflow every time a new pull request is created in the repository"

It looks like it will just happen during a PR creation. 
Otherwise all look good to me.

Reply:
What's the reason for triggering the test workflow upon push events?

Reply:
The motivation is that when we need to force-push to a branch, we hope to trigger this workflow too.

Reply:
IMO, we should never allow force push to an official branch such as 2.4-li.
Assuming there are no force-pushes, then it seems we don't need to trigger CI on regular push events.
I'd suggest we remove these, what do you think?